### ISSUE TITLE
Cannot trace block #16292612 using callTracer

### ISSUE THREAD

**Issue by cp0k** [celo-org/celo-blockchain#1983]
## Description

The need here is to trace block #16292612 while calling on method `debug_traceBlockByNumber` using the Tracer `callTracer`

## Current Behavior

➜  ~ curl https://RPC_PROVIDER/ \ 
 -X POST \                                                                       
 -H "Content-Type: application/json" \
 --data '{"method":"debug_traceBlockByNumber","params":["0xf89b04", {"tracer": "callTracer"}],"id":1,"jsonrpc":"2.0"}'  | jq '.result[93]'

{
  "error": "tracing failed: max fee per gas less than block base fee: address 0xadd7c542667728e6B91Dc091a6def2489f2dE35C, maxFeePerGas: 227247290 baseFee: 229157277"
}
## Desired Behavior

To see the trace results and not the failure I pasted above.

## Context

Operating System: "Ubuntu 20.04.5 LTS (Focal Fossa)"
Client Version: celo/v1.7.0-stable-5285ca48/linux-amd64/go1.13.8

## Proposed Solution

"Pending"

**Comment by gastonponti** [comment]
Hi @cp0k!
TL;DR; The PR #1982 should fix this issue.

The problem is in the tracer.

We use a Block Context (Part of the Espresso HF) at the beginning of the block, to retrieve the rates of all the stables.
That rate is the one used to order and validate the txs that are going to be part of the block. Using this context allow us to avoid reshuffling all the mempool after each tx apply because the applied tx could change the rate of a stable (this is explained in the Espresso HF page).
In this case, that block, has a tx that changes the rate of the cUSD, and the last tx that is trying to execute, uses the cUSD as the currencyFee, and the new rate is bigger that the fee of that transaction.
This happens because the tracer was generating a Block Context after every tx execution, and not sharing the same context created at the beginning of the block.

The PR #1987 should fix this error.
